src/js_types.hpp

```diff
@@ -616,8 +619,7 @@ inline typename T::Value Value<T>::from_bson(typename T::Context ctx, const bson
         // it to a plain js number if it is in the range where it can be done precisely, otherwise
         // we map to the bson.Long type which preserves the value, but is harder to use.
         const auto i64_val = value.operator int64_t();
-        constexpr static int64_t max_precise_double = 1ll << 52; // 52 bits mantissa + implicit leading 1 bit.
-        if (-max_precise_double <= i64_val && i64_val <= max_precise_double)
+        if (-JS_MAX_SAFE_INTEGER <= i64_val && i64_val <= JS_MAX_SAFE_INTEGER)
```

Comment:
Does JS have an actual definitoin of "min safe integer"  (these are usually something like `-JS_MAX_SAFE_INTEGER - 1`)?

Reply:
Yes `-Number.MAX_SAFE_INTEGER === Number.MIN_SAFE_INTEGER` => `true`.

Reply:
Great